File: atomic/src/main/java/com/infideap/atomic/ResponseFutureCallback.java
package com.infideap.atomic;

import okhttp3.Response;

/**
 * Created by Shiburagi on 20/10/2016.
 */
public class ResponseFutureCallback<T> implements FutureCallback<T> {
    public void onCompleted(Exception e, T result) {

    }

    public void onCompleted(Exception e, T result, Response response) {
        onCompleted(e, result);
    }

    ;
}

File: atomic/src/main/java/com/infideap/atomic/ResponseFutureCallback.kt
package com.infideap.atomic

import okhttp3.Response

/**
 * Created by Shiburagi on 20/10/2016.
 */
class ResponseFutureCallback<T> : FutureCallback<T> {
    override fun onCompleted(e: Exception?, result: T) {}
    fun onCompleted(e: Exception?, result: T, response: Response?) {
        onCompleted(e, result)
    }
}